### ISSUE TITLE
Texts on major ticks are very pixelated

### ISSUE THREAD

**Issue by johanolofsson** [oxyplot/oxyplot#1720]
### Steps to reproduce

1. Create any plot with major ticks and zoom in.

Platform:
.NET Core 3.1 WPF

### Expected behaviour

The text (numbers) should be rendered smoothly (just like annotations).

### Actual behaviour

The text looks very pixelated. As if it was rendered in low resolution and then resized.

![image](https://user-images.githubusercontent.com/4460153/100072796-fba8ef00-2e3c-11eb-8603-ca5a4ad1622b.png)

**Comment by VisualMelon** [comment]
What do you mean by zooming it? E.g. using a magnifier? I can't immediately reproduce this: can you provide a complete example? Could you verify which version of the oxyplot library you are using? (i.e. 2.0 or 2.1-Preview1, and if the preview, Canvas or Skia?)

They look like sub-pixel rendering artefacts somehow made worse (maybe by the shader? is it definitely applied to the whole plot?). The rendering routines for axis labels and annotations are all the same, so there shouldn't be any difference. Sub-pixel rendering is enabled for all text.

**Comment by johanolofsson** [comment]
Thanks for your reply!

* The zoom behavior should be able to mimic just by creating a very large plot and place in a scrollviewer. 
* We are using OxyPlot.Wpf (2.0.0) CanvasRenderer.

Yes indeed the pixel shader enhances the issue and like you say it's likely a subpixel related issue. However the same issue is there, altough not nearly as obvious, without our shader.

I'll attach an image which is a cut out of the application (without shader) and then enlarged in paint.net.

As you can see there are clearly something different going on around the the major tick text 3 compared to the annotation. Like a red aura on the left of the silhouette and a blue on the right.

![pixelated2](https://user-images.githubusercontent.com/4460153/100079146-9bb64680-2e44-11eb-8fc7-71951e311baa.png)

**Comment by VisualMelon** [comment]
Yeah, that's classic sub-pixel rendering on the right: must be disabled for some reason on the right; I can't think why they are different... they are splatted onto the same layer with the same properties, but I'd better check the 2.0 sources (thanks for those details). I don't have time to look into this properly today, but I'll try to pick it up tomorrow: feel free to pester me if nobody gets back to you.

**Comment by johanolofsson** [comment]
Sounds great, we're not in hurry to get this fixed so don't stress it!

Thanks for a your great work, OxyPlot has surely been awesome for us!

**Comment by johanolofsson** [comment]
Btw, we just noticed the same issue applies to legend texts.

**Comment by VisualMelon** [comment]
I'm still not able to reproduce this, or come up with a mechanism by which the axis text should be rendered differently from an annotation: are the annotations definitely provided by OxyPlot and not component? Can you check in the live-visual-tree that the TextBlocks have the same properties and all existing within the `PlotView` element?

Apparently WPF will disable ClearType in some circumstances, but I can't think why that would be the case for the annotation (though I'm not particularly familiar with WPF).

Would it be possible for you to supply a reproducing example? Don't worry about making it minimal if it's a pain (assuming there are no privacy concerns).

That said, I suspect the only solution that will work for you would be to disable ClearType altogether on the plot, something `CanvasRenderContext` doesn't support (though it should be easy to implement); rather, it _explicitly_ sets the `ClearTypeHint` to `Enabled`, which makes it difficult to override as the user.

**Comment by johanolofsson** [comment]
Ok, forcing the clear type to Auto fixed the issue. It was quite easy to apply since we already had a custom render context from previously patched issues. Just get the canvas using reflection and apply it to the last child after each DrawText call.
